runtime/thunk/thunk.ts

```diff
@@ -133,6 +138,47 @@ async function doFindAndExecutePackFunction<T extends FormulaSpecification>({
             return parentFormula.execute(params as any, executionContext);
           }
           break;
+        case MetadataFormulaType.PropertyAutocomplete:
+          const syncTable = syncTables?.find(table => table.name === formulaSpec.syncTableName);
+          const autocompleteFn = ensureExists(syncTable?.propertyAutocomplete);
+          const propertyValues = {};
+
+          const cacheKeysUsed: string[] = [];
+
+          function recordPropertyAccess(key: string) {
+            if (!cacheKeysUsed.includes(key)) {
+              cacheKeysUsed.push(key);
+            }
+          }
+
+          for (const [key, value] of Object.entries(formulaSpec.propertyValues)) {
+            Object.defineProperty(propertyValues, key, {
+              get() {
+                recordPropertyAccess(key);
+                return value;
+              },
+            });
+          }
+
+          const cellAutocompleteExecutionContext: Omit<PropertyAutocompleteExecutionContext, 'search'> = {
+            ...executionContext,
+            propertyName: formulaSpec.propertyName,
+            propertyValues,
+          };
+
+          Object.defineProperty(cellAutocompleteExecutionContext, 'search', {
+            get() {
+              recordPropertyAccess('__search');
```

Comment:
Wonder if __search should be a named constant, since presumably you're going to reference it on the `coda` side?

Reply:
Oh, I forgot I'd done that. The one other place this constant appears in the prototype is strictly for implementing the caching.

Another option here could be to record the property access in a totally different way so the output looks more like `{results: [...], propertiesUsed: ['foo'], searchUsed: true}`

The cache implementation might still want to try to use strings as cache keys, but then the problem of converting this to and from a string would be localized there.

There are other things that could impact caching that aren't implemented here (like a change to dynamic schema) but I suspect the easiest way to deal with those is to just clear the cache for any sync table if the connection or schema changes.

Reply:
Updated the code to use a dedicated property to indicate if the search string was read instead of a magic constant within `propertiesUsed`